packages/jest-preset-default/jest-preset.json

```diff
@@ -27,5 +27,6 @@
 	"transform": {
 		"^.+\\.[jt]sx?$": "<rootDir>/node_modules/babel-jest"
 	},
-	"verbose": true
+	"verbose": true,
```

Comment:
With the latest changes, I wonder if this line is even necessary. And then, it gets me considering, if someone was to use this preset as a base, and then override `"verbose": false`, would it actually disable the `VerboseRunner` from being used?

It seems like something where the additional reporter shouldn't be making any assumptions outside its own custom behavior, which we could do if we included `"default"` in the reporters array:

https://jestjs.io/docs/en/configuration#reporters-arraymodulename--modulename-options

I suppose the problem then is: What would the export of `travis-folder-passes.reporter.js` be in the `else` condition of `'TRAVIS' in process.env` ?

I could see a few options:

- Move the condition into each of the `onTestResult` and `onRunComplete` to treat them as a noop.
- Maybe it's enough to revert back to extending `DefaultReporter` if we include `'default'` in the `reporters` configuration?
- Or, we could convert this file to a JavaScript file, and conditionally include the custom reporter path based on the `process.env` condition.

Reply:
>Move the condition into each of the onTestResult and onRunComplete to treat them as a noop.

We still need to extend either the verbose or default reporter to get some output without running 2 reporters in Travis.

>Maybe it's enough to revert back to extending DefaultReporter if we include 'default' in the reporters configuration?

Likewise, that would run the 2 reporters.

>Or, we could convert this file to a JavaScript file, and conditionally include the custom reporter path based on the process.env condition.

That would work.

A potentially easier solution that respects the actual configuration would be to read the current Jest config, which reporters are constructed with:

https://github.com/facebook/jest/blob/ad5377333daf6716af3465bba39f86b7db485e2b/packages/jest-reporters/src/verbose_reporter.ts#L25

Reply:
Seems to work: https://github.com/WordPress/gutenberg/pull/16755/commits/f7c72e76efef6c4f53eb609df035f4818f5ec87c.

Reply:
I think it can work. It still doesn't quite sit well with me, since it makes assumptions about what the default behavior would be from our custom code, rather than just letting Jest deal with the defaults. Converting `jest-preset.json` to a JavaScript file and manipulating the `reporters` array feels like a more durable long-term solution. I'm fine to explore that separately.

Reply:
Agreed.

Reply:
#18896